crates/ty_project/src/db.rs

```diff
@@ -37,7 +37,19 @@ impl ProjectDatabase {
     {
         let mut db = Self {
             project: None,
-            storage: salsa::Storage::default(),
+            storage: salsa::Storage::new(Some(Box::new({
+                move |event| {
+                    if !tracing::enabled!(tracing::Level::TRACE) {
+                        return;
+                    }
+
+                    if matches!(event.kind, salsa::EventKind::WillCheckCancellation) {
+                        return;
+                    }
+
+                    tracing::trace!("Salsa event: {event:?}");
+                }
+            }))),
```

Comment:
`tracing::enabled` shouldn't change during a run. Therefore, it would be better to set the event handler to `None` if the `trace` level isn't enabled.

Reply:
Done. I tried a few variations because I wanted to pull it out into a variable (and potentially use `bool::then`), but that requires a complex type annotations, so I just left it inline.